spring-cloud-kubernetes-controllers/spring-cloud-kubernetes-configuration-watcher/src/main/java/org/springframework/cloud/kubernetes/configuration/watcher/RefreshTrigger.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.configuration.watcher;
+
+import io.kubernetes.client.common.KubernetesObject;
+import reactor.core.publisher.Mono;
+
+/**
+ * Defines the refresh trigger contract.
+ *
+ * @author wind57
+ */
+interface RefreshTrigger {
```

Comment:
one of the main ideas of this PR is here. So basically, we have two ways to refresh : bus based and event based. If you look closely at bus one (for example), then both secrets and configmaps implementations are doing the same thing, with slightly different arguments (for example they pass a different label to be checked). So my idea is: why not unify implementations? And also add this abstraction so that the code becomes more readable and made more sense?

Reply:
So I delegate to two places for events triggering : `BusRefreshTrigger` and `HttpRefreshTrigger`. It also helps a lot that kubernetes client has a common class called `KubernetesObject` that both `V1ConfigMap` and `V1Secret` implements.